File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    answer = 1
    targets.sort(key=lambda x:x[1])
    h = targets[0][1]
    for i in range(1,len(targets)):
        if h > targets[i][0]:
            continue
        else:
            h = targets[i][1]
            answer+=1
    return answer

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        int last_start = 0;
        Arrays.sort(targets, Comparator.comparingInt((int[] target) -> target[1]));
        
        for(int[] target : targets){
            int start = target[0];
            int end = target[1];
    
            if(start >= last_start){
                answer++;
                last_start=end;
            }
            
        }

        return answer;
    }
}